command/edit.go

```diff
@@ -0,0 +1,136 @@
+package command
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+
+	cli "gopkg.in/urfave/cli.v1"
+	log "github.com/Sirupsen/logrus"
+	yaml "gopkg.in/yaml.v2"
+	vault "github.com/seatgeek/hashi-helper/command/vault"
+)
+
+type preferences map[string]preference
+type preference struct {
+	KeybaseTeam  string `yaml:"keybase-team-name"`
+	Editor string `yaml:"editor"`
+}
+
+// EditEncryptedFile ...
+func EditEncryptedFile(c *cli.Context) error {
+
+	// identify the target file
+	if !c.Args().Present() {
+		return fmt.Errorf("Please provide a path and file name.")
+	}
+	filePath := c.Args().First()
+	if filePath == "" {
+		return fmt.Errorf("Empty path and file name.")
+	}
+
+	// get the preferences, define the editor and keybase team
+	var preferences preferences
+	configFile, err := vault.GetProfileConfig()
+	if err != nil {
+		return err
+	}
+	if err := yaml.Unmarshal(configFile, &preferences); err != nil {
+		return err
+	}
+	preference, ok := preferences["encrypted-file-config"]
+	if !ok {
+		return fmt.Errorf("No encrypted-file-config block found in config file.\n")
+	}
+	// editor
+	editor := os.Getenv("EDITOR")
+	if editor == "" {
+		editor = preference.Editor
+	}
+	if editor == "" {
+		editor = "pico"
+	}
+	// keybase team
+	var keybaseTeam string = preference.KeybaseTeam
+	if keybaseTeam == "" && NoKeybaseTeam == false {
+		log.Warnf("No Keybase Team found in config (and --no-keybase-team is unset). Not using Keybase Team.\n")
```

Comment:
this might be desired, not sure if it should be a warning? 

If its not defined in any configuration source, the desired behaviour might be to use local keybase instead

Reply:
Good thought. I was only thinking of my particular use case (signing with multiple PGP keys [a team] is required). But in general I suppose that is not the case. Combined with your comment above w.r.t. `KEYBASE_TEAM`, I agree that will make for a more sane set of default behaviors.

Reply:
Good thought. I was only thinking of my particular use case (signing with multiple PGP keys [a team] is required). But in general I suppose that is not the case. Combined with your comment above w.r.t. KEYBASE_TEAM, I agree that will make for a more sane set of default behaviors.